src/interface/src/app/services/auth.service.ts

```diff
@@ -322,3 +353,28 @@ export class ValidationResolver implements Resolve<boolean> {
     );
   }
 }
+
+/** Resolver to validate the password reset token. */
+export const passwordResetTokenResolver: ResolveFn<
```

Comment:
Nice! today I've learned that https://angular.io/api/router/Resolve is deprecated in favor of this solution.
My only comment is that this might be better in its own file

Reply:
Done, added a password-reset.ts file.